src/client/fontengine.cpp

```diff
@@ -18,10 +18,11 @@
 /** reference to access font engine, has to be initialized by main */
 FontEngine *g_fontengine = nullptr;
 
-/** callback to be used on change of font size setting */
-static void font_setting_changed(const std::string &name, void *userdata)
+bool FontEngine::m_needs_reload = false;
+
+void FontEngine::fontSettingChanged(const std::string &name, void *userdata)
 {
-	static_cast<FontEngine *>(userdata)->readSettings();
+	m_needs_reload = true;
```

Comment:
A static member makes this code messy. You can access it within the `FontEngine` class by using `userdata` as follows:
```C++
((FontEngine *)userdata)->m_needs_reload = true;
```